WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,617 @@
+    }
+    
+    private func noResultsTitles(draft: String, scheduled: String, trashed: String, published: String) -> [PostListStatusFilter:String] {
+        return [.Draft: draft,
+                .Scheduled: scheduled,
+                .Trashed: trashed,
+                .Published: published]
+    }
+    
+    func noResultsMessageText() -> String {
+        if syncHelper?.isSyncing == true || isSearching() {
+            return ""
+        }
+        
+        let filter = currentPostListFilter()
+        
+        // currentPostListFilter() may return `nil` at this time (ie: it's been declared as
+        // `nullable`).  This will probably change once we can migrate
+        // AbstractPostListViewController to Swift, but for the time being we're defining a default
+        // filter here.
+        //
+        // Diego Rey Mendez - 2016/04/18
+        //
+        let filterType = filter?.filterType ?? .Draft
+        var message : String
+        
+        switch filterType {
+        case .Draft:
+            message = NSLocalizedString("Would you like to create one?", comment: "Displayed when the user views drafts in the pages list and there are no pages")
+            break
+        case .Scheduled:
+            message = NSLocalizedString("Would you like to schedule a draft to publish?", comment: "Displayed when the user views scheduled pages in the oages list and there are no pages")
+            break
+        case .Trashed:
+            message = NSLocalizedString("Everything you write is solid gold.", comment: "Displayed when the user views trashed pages in the pages list and there are no pages")
+            break
+        default:
+            message = NSLocalizedString("Would you like to publish your first page?", comment: "Displayed when the user views published pages in the pages list and there are no pages")
+            break
+        }
+        
+        return message
+    }
+    
+    
+    func noResultsButtonText() -> String? {
+        if syncHelper?.isSyncing == true || isSearching() {
+            return nil
+        }
+        
+        let filter = currentPostListFilter()
+        
+        // currentPostListFilter() may return `nil` at this time (ie: it's been declared as
+        // `nullable`).  This will probably change once we can migrate
+        // AbstractPostListViewController to Swift, but for the time being we're defining a default
+        // filter here.
+        //
+        // Diego Rey Mendez - 2016/04/18
+        //
+        let filterType = filter?.filterType ?? .Draft
+        var title : String
+        
+        switch filterType {
+        case .Trashed:
+            title = ""
+            break
+        default:
+            title = NSLocalizedString("Start a Page", comment: "Button title, encourages users to create their first page on their blog.")
+            break
```

Comment:
Same notes as above:

- Break is not needed
- Maybe return the right string, riiight there?